src/figuresClasses.ts

```diff
@@ -1,11 +1,75 @@
-export interface Figure {}
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [this.a, this.b, this.c].sort((x, y) => y - x);
+    const [longest, side1, side2] = sides;
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0 || longest >= side1 + side2) {
+      throw new Error('your error message');
```

Comment:
The error message 'your error message' is not descriptive. Consider providing a more informative message, such as 'Invalid triangle dimensions'.